classes/navigationMenu/NavigationMenuDAO.inc.php

```diff
@@ -0,0 +1,303 @@
+		}
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Retrieve a navigation menu by context Id.
+	 * @param $contextId int Context Id
+	 * @return NavigationMenu
+	 */
+	function getByContextId($contextId) {
+		$params = array((int) $contextId);
+		$result = $this->retrieve(
+			'SELECT * FROM navigation_menus WHERE context_id = ?',
+			$params
+		);
+
+		return new DAOResultFactory($result, $this, '_fromRow');
+	}
+
+	/**
+	 * Retrieve a navigation menu by navigation menu ID.
+	 * @param $contextId int Context Id
+	 * @param $areaName string Template Area name
+	 * @return NavigationMenu
+	 */
+	function getByArea($contextId, $areaName) {
+		$params = array($areaName);
+		$params[] = (int) $contextId;
+		$result = $this->retrieve(
+			'SELECT * FROM navigation_menus WHERE area_name = ? and context_id = ?',
+			$params
+		);
+
+		$returner = null;
+		if ($result->RecordCount() != 0) {
+			$returner = $this->_fromRow($result->GetRowAssoc(false));
+		}
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Retrieve a navigation menu by title
+	 * @param $contextId int Context Id
+	 * @param $title string
+	 * @return NavigationMenu
+	 */
+	function getByTitle($contextId, $title) {
+		$params = array((int) $contextId);
+		$params[] = $title;
+		$result = $this->retrieve(
+			'SELECT * FROM navigation_menus WHERE context_id = ? and title = ?',
+			$params
+		);
+
+		$returner = null;
+		if ($result->RecordCount() != 0) {
+			$returner = $this->_fromRow($result->GetRowAssoc(false));
+		}
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Check if a navigationMenu exists with the given title.
+	 * @param $title int
+	 * @param $contextId int
+	 * @return boolean
+	 */
+	function navigationMenuExistsByTitle($title, $contextId) {
+		$result = $this->retrieve(
+			'SELECT COUNT(*)
+			FROM	navigation_menus
+			WHERE	title = ? AND
+				context_id = ?',
+			array(
+				$title,
+				(int) $contextId
+			)
+		);
+		$returner = isset($result->fields[0]) && $result->fields[0] != 0 ? true : false;
+
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Get the locale field names.
+	 * @return array
+	 */
+	function getLocaleFieldNames() {
+	    return array();
+	}
+
+	/**
+	 * Internal function to return an NavigationMenu object from a row.
+	 * @param $row array
+	 * @return NavigationMenu
+	 */
+	function _fromRow($row) {
+		$navigationMenu = $this->newDataObject();
+		$navigationMenu->setId($row['navigation_menu_id']);
+		$navigationMenu->setTitle($row['title']);
+		$navigationMenu->setAreaName($row['area_name']);
+		$navigationMenu->setContextId($row['context_id']);
+		$navigationMenu->setDefault($row['is_default']);
+
+		return $navigationMenu;
+	}
+
+	/**
+	 * Insert a new NavigationMenu.
+	 * @param $navigationMenu NavigationMenu
+	 * @return int
+	 */
+	function insertObject($navigationMenu) {
+		$this->update(
+				'INSERT INTO navigation_menus
+				(title, area_name, context_id, is_default)
```

Comment:
If this is a symbolic name, then perhaps it should be called `symbolic` instead, and we could use `CAPITALIZED_VALUES` for it. These might be useful handles to use throughout e.g. in case we want to have a way for upgrade scripts to alter default menu items without having to wrestle with finding localized names and hoping for the best.